### PR TITLE
Breaking: Separate circuit, burner and compressor and add hybrid device

### PR DESCRIPTION
**Pull request by woehrl01**
We currently only "use" circuits. The problem is that there is a difference between `circuits`, `burners` and `compressors`. That means that we mix the property of all these up.

This PR fixes that, by introducing proper separation between these parts of the device.

* Adds hybrid device as discussed in #175 
* Add all missing properties, existing in response files
* Fixed `heating.power.consumption.total` due to API change
* Added handler for 500 status code errors (see #179)
* Added unittest for new properties

### CODE REVIEW

File: PyViCare/PyViCareGazBoiler.py [hunk 1]
```diff
@@ -112,16 +122,20 @@ def getPowerConsumptionThisYear(self):
...
 
-class GazBoilerWithCircuit(DeviceWithCircuit):
+class GazBurner(DeviceWithComponent):
+
+    @property
+    def burner(self) -> str:
+        return self.component
 
     @handleNotSupported
     def getBurnerHours(self):
```
Comment (crazyfx1) [inline PyViCare/PyViCareGazBoiler.py]: We could think about renaming them to just "getHours()", "getStarts()".

File: PyViCare/PyViCareHeatPump.py [hunk 2]
```diff
@@ -20,36 +30,40 @@ def getReturnTemperaturePrimaryCircuit(self):
...
 
-class HeatPumpWithCircuit(DeviceWithCircuit):
+class Compressor(DeviceWithComponent):
+
+    @property
+    def compressor(self) -> str:
+        return self.component
 
     @handleNotSupported
     def getCompressorStarts(self):
```
Comment (crazyfx1) [inline PyViCare/PyViCareHeatPump.py]: Same here, "getStarts()", "getHours()"

**Review by crazyfx1 — APPROVED** [review]
Looks great :)

File: PyViCare/PyViCareHeatPump.py (hunk 2 above)
Comment (woehrl01) [reply PyViCare/PyViCareHeatPump.py]: great idea 👍🏼

**Pull request merged**